scripts/es-integration-test.sh

```diff
@@ -19,114 +19,88 @@ check_arg() {
-    --env "http.host=0.0.0.0"
-    --env "transport.host=127.0.0.1"
-    --env "xpack.security.enabled=false"
-  )
-  local major_version=${tag%%.*}
-  if (( major_version < 8 )); then
-    params+=(--env "xpack.monitoring.enabled=false")
-  else
-    params+=(--env "xpack.monitoring.collection.enabled=false")
-  fi
-  if (( major_version > 7 )); then
-    params+=(
-      --env "action.destructive_requires_name=false"
-    )
-  fi
-
-  local cid
-  cid=$(docker run "${params[@]}" "${image}:${tag}")
+  local compose_file=$1
+  docker compose -f "${compose_file}" up -d
+  local cid 
+  cid=$(docker compose -f "${compose_file}" ps -q elasticsearch)
   echo "cid=${cid}" >> "$GITHUB_OUTPUT"
-  echo "${cid}"
 }
 
 setup_opensearch() {
-  local image=opensearchproject/opensearch
-  local tag=$1
-  local params=(
-    --detach
-    --publish 9200:9200
-    --env "http.host=0.0.0.0"
-    --env "transport.host=127.0.0.1"
-    --env "plugins.security.disabled=true"
-  )
-  local cid
-  cid=$(docker run "${params[@]}" "${image}:${tag}")
+  local compose_file=$1
+  compose_file="docker-compose/opensearch/v${major_version}.yml"
+  docker compose -f "${compose_file}" up -d
+  local cid 
+  cid=$(docker compose -f "${compose_file}" ps -q opensearch)
   echo "cid=${cid}" >> "$GITHUB_OUTPUT"
-  echo "${cid}"
 }
 
 wait_for_storage() {
   local distro=$1
   local url=$2
-  local cid=$3
+  local compose_file=$3
   local params=(
     --silent
     --output
     /dev/null
     --write-out
     "%{http_code}"
   )
-  local counter=0
-  local max_counter=60
-  while [[ "$(curl "${params[@]}" "${url}")" != "200" && ${counter} -le ${max_counter} ]]; do
-    docker inspect "${cid}" | jq '.[].State'
-    echo "waiting for ${url} to be up..."
+  local max_attempts=60
+  local attempt=0
+  echo "Waiting for ${distro} to be available at ${url}..."
+  until [[ "$(curl "${params[@]}" "${url}")" == "200" ]] || (( attempt >= max_attempts )); do
+    echo "Attempt $(( attempt + 1 )): ${distro} is not yet available at ${url}..."
     sleep 10
-    counter=$((counter+1))
+    attempt=$(( attempt + 1 ))
   done
-  # after the loop, do final verification and set status as global var
+
   if [[ "$(curl "${params[@]}" "${url}")" != "200" ]]; then
-    echo "ERROR: ${distro} is not ready"
-    docker logs "${cid}"
-    docker kill "${cid}"
+    echo "ERROR: ${distro} is not ready at ${url} after $(( attempt * 10 )) seconds"
+    docker compose -f "${compose_file}" logs
+    docker compose -f "${compose_file}" down
     db_is_up=0
   else
-    echo "SUCCESS: ${distro} is ready"
+    echo "SUCCESS: ${distro} is available at ${url}"
     db_is_up=1
   fi
 }
 
 bring_up_storage() {
   local distro=$1
   local version=$2
-  local cid
+  local major_version=${version%%.*}
+  local compose_file="docker-compose/${distro}/v${major_version}.yml"
 
   echo "starting ${distro} ${version}"
   for retry in 1 2 3
   do
     echo "attempt $retry"
     if [ "${distro}" = "elasticsearch" ]; then
```

Comment:
I don't think this bifurcation is necessary any longer. There is one place where string like "elasticsearch" is used in the function (when getting container ID from docker by name), so you can pass distro name as argument and merge setup into a single `setup_db` function